sklearn/cluster/tests/test_mean_shift.py

```diff
@@ -39,27 +38,34 @@ def test_estimate_bandwidth():
     assert 0.9 <= bandwidth <= 1.5
 
 
-def test_estimate_bandwidth_1sample():
+def test_estimate_bandwidth_1sample(global_dtype):
     # Test estimate_bandwidth when n_samples=1 and quantile<1, so that
     # n_neighbors is set to 1.
-    bandwidth = estimate_bandwidth(X, n_samples=1, quantile=0.3)
-    assert bandwidth == pytest.approx(0.0, abs=1e-5)
+    bandwidth = estimate_bandwidth(
+        X.astype(global_dtype, copy=False), n_samples=1, quantile=0.3
+    )
+
+    assert bandwidth.dtype == X.dtype
+    assert_allclose(bandwidth, 0.0, atol=1e-5)
 
 
 @pytest.mark.parametrize(
     "bandwidth, cluster_all, expected, first_cluster_label",
     [(1.2, True, 3, 0), (1.2, False, 4, -1)],
 )
-def test_mean_shift(bandwidth, cluster_all, expected, first_cluster_label):
+def test_mean_shift(
+    global_dtype, bandwidth, cluster_all, expected, first_cluster_label
+):
     # Test MeanShift algorithm
+    X_ = X.astype(global_dtype, copy=False)
     ms = MeanShift(bandwidth=bandwidth, cluster_all=cluster_all)
-    labels = ms.fit(X).labels_
+    labels = ms.fit(X_).labels_
     labels_unique = np.unique(labels)
     n_clusters_ = len(labels_unique)
     assert n_clusters_ == expected
     assert labels_unique[0] == first_cluster_label
 
-    cluster_centers, labels_mean_shift = mean_shift(X, cluster_all=cluster_all)
+    cluster_centers, labels_mean_shift = mean_shift(X_, cluster_all=cluster_all)
     labels_mean_shift_unique = np.unique(labels_mean_shift)
```

Comment:
`cluster_centers` is not used here. I think that we can add a test for the `dtype`.